packages/blade/src/components/Tag/_decisions/decisions.md

```diff
@@ -0,0 +1,198 @@
+- `valuePrefix` (Similar to `titleSuffix` from BaseHeader)
+
+<details>
+<summary>Full Controlled Example</summary>
+
+```jsx
+function App() {
+  const [inputValue, setInputValue] = React.useState('');
+  const [tags, setTags] = React.useState([]);
+
+  const addTag = () => {
+    // Add input value to tags and clear the input value
+    setTags([...tags, inputValue]);
+    setInputValue('');
+  }
+
+  const removeTag = (tagName) => {
+    setTags(tags.filter(tagNameValue) => tagNameValue !== tagName);
+  }
+
+  return (
+    <TextInput
+      tagsSlot={tags.map((tagName, index) => (
+        <Tag onDismiss={({ value }) => removeTag(value)}>{tagName}</Tag>
+      ))}
+      value={inputValue}
+      onChange={({ value }) => setInputValue(value)}
+      onKeyDown={(e) => {
+        if (e.key === 'ENTER') {
+          addTag();
+        }
+      }}
+    />
+  );
+}
+```
+
+</details>
+
+<details>
+<summary>Alternate Approach</summary>
+
+#### Alternate Approach: `value` prop
+
+We can extend our `value` prop to accept JSX
+
+```jsx
+<TextInput
+  value={
+    <>
+      <Tag onDismiss={}>kamlesh.chandnani@razorpay.com</Tag>
+      <Tag onDismiss={}>divyanshu.maithani@razopay.com</Tag>
+      saurabhdaw
+    </>
+  }
+  onChange={}
+/>
+```
+
+**Cons**
+
+- Requires us to loop on children and separate out Tags from written word and this will run on every `onChange` event.
+- Have not seen any other library handle it this way
+
+</details>
+
+## Accessibility
+
+- Ensure Close Icon is focussable
+- Close Icon to have appropriate `aria-label`
+- Special keyboard interaction when used inside Input components
+  - Pressing `Tab` should move focus into the input and not Tag
+  - Tag is focussable with Left and Right Arrow Keys
+  - Pressing `Backspace` on Tag should remove it
+  - Clicking on Tag should bring focus on Input
+- Tag should be disabled when used inside a disabled input
+
+Refer to Keyboard Interactions of [react-select](https://react-select.com/home#getting-started) for reference. Other Input components with Tag also follow similar keyboard interactions.
+
+## References
+
+- [react-select](https://react-select.com/home#getting-started)
+- [`TagsInput` component by PluralSight Design System](https://design-system.pluralsight.com/components/tagsinput)
+- [`tokens` prop by Primer AutoComplete.Input](https://primer.style/react/Autocomplete/)
+- [Carbon just putting everything outside of Inputs](https://carbondesignsystem.com/components/tag/usage)
+- [`elemBeforeInput` prop on Atlassian TextField](https://atlassian.design/components/textfield/examples#elements-before-and-after-input)
+
+## Open Questions
+
+- Do we want to discuss Tag's integration with TextInput and TextArea? or should I remove it from API decision altogether?
+  - We can discuss here
+- ~`<Tag />` or `<Tags />` 😅~ `<Tag />`
+- [`tagsSlot` vs `tagSlot` vs `valuePrefix` vs `leading` on Input components](#alternate-prop-name)
```

Comment:
If we're adding it to consumer-facing TextInput, we could name it `_tagsSlot` and throw warnings if a consumer tries to use it

Reply:
It should be allowed for consumers as well. You can check the TextArea example here - https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=29212%3A571254&mode=dev

Reply:
what if we just call it as tags(that accepts array of tag)? since its literally that?

Reply:
By array of tags you mean something like `tags={['apple', 'banana']}`? or `tags={[<Tag>apple</Tag>, <Tag>banana</Tag>]}`

We can't do 1st one because we need `onDismiss` of Tag exposed as well so that closing it can remove that tag.

Used `tagsSlot` communicates that it's a slot where Tag components can be passed. But I think `tags={<Tag>apple</Tag><Tag>banana</Tag>}` also makes sense.

Reply:
meant this the prop name `tags` and the value as `tags={[<Tag>apple</Tag>, <Tag>banana</Tag>]}`

wouldn't this `tags={<Tag>apple</Tag><Tag>banana</Tag>}` be more difficult to type? like people misusing the jsx thing? with array of jsx you can at least type what typeof array elements we are expecting.

Reply:
`tags` prop name makes sense. For TS, I was basically doing this `tags={<><Tag>apple</Tag><Tag>banana</Tag></>}`. I think what you mentioned can be done as well. We'll get to add some spacing between array items while rendering with array syntax. With jsx we might have to loop over with React.Children.